include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -583,7 +1098,8 @@ template <typename Lyt, typename TT>
 [[nodiscard]] std::pair<operational_status, std::size_t>
 is_operational(const Lyt& lyt, const std::vector<TT>& spec, const is_operational_params& params = {},
                const std::optional<std::vector<bdl_wire<Lyt>>>& input_bdl_wire  = std::nullopt,
-               const std::optional<std::vector<bdl_wire<Lyt>>>& output_bdl_wire = std::nullopt)
+               const std::optional<std::vector<bdl_wire<Lyt>>>& output_bdl_wire = std::nullopt,
+               const std::optional<Lyt>&                        canvas_lyt      = std::nullopt) noexcept
```

Comment:
With all these optional parameters, it seems to me that there is something wrong with the usability of this function. It should probably be (at least) two functions at this point.

Reply:
I'll have a detailed look at all the changes in this file once we figure out the API.